### DISCUSSION TITLE
Added Collaborator with commits not showing as Contributor

### DISCUSSION THREAD

**Discussion by 0deadLock0** [community/community#23379] [New to GitHub]
<p><strong>TL;DR</strong></p>
<blockquote>
<p>A person’s commits made can be seen with their GitHub ID in the commit history and the person is also added as a collaborator. Yet the person is not showing as a Contributor.</p>
</blockquote>
<p>I have a  GitHub repository created by me, which was private until now. To allow my friend to commit to it, I had added him as Collaborator (in the “Manage Access” section).</p>
<p>After getting access to the then private repository he committed several times. But, his name never got into the contributors. It felt maybe, of the repository being private.</p>
<p>But now when the repository is turned public, his name is yet not included in the contributors.</p>
<p>Note: Both of us are relatively new to GitHub.</p>

**Comment by canuckjacq** [comment, marked as answer]
<p>I’ve had a look - I can see that there is one contributor on one of your repositories who may have their email address stored with a typo in their git config settings. This would lead to their commits not appearing to come from their account, and therefore not listed as a contributor.</p>
<p>Your friend can check quickly by using this command:</p>
<pre><code> git config --global user.email
</code></pre>
<p>If it is wrong, this guide may help:</p>
<aside class="onebox githubfolder">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/free-pro-team@latest/github/setting-up-and-managing-your-github-user-account/setting-your-commit-email-address#setting-your-commit-email-address-in-git" target="_blank" rel="noopener">docs.github.com</a>
  </header>
  <article class="onebox-body">
    

<h3><a href="https://docs.github.com/en/free-pro-team@latest/github/setting-up-and-managing-your-github-user-account/setting-your-commit-email-address#setting-your-commit-email-address-in-git" target="_blank" rel="noopener">Setting your commit email address - GitHub Docs</a></h3>

<p><a href="https://docs.github.com/en/free-pro-team@latest/github/setting-up-and-managing-your-github-user-account/setting-your-commit-email-address#setting-your-commit-email-address-in-git" target="_blank" rel="noopener">//docs.github.com/en/free-pro-team@latest/github/setting-up-and-managing-your-github-user-account/setting-your-commit-email-address</a></p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Reply by Busaleem** [reply]
I think

**Comment by 0deadLock0** [comment]
<p>Thanks, indeed this was the issue.</p>
<p>But, I wonder when someone is providing their GitHub credentials accurately why do GitHub specifically need the email address.</p>
<p>Isn’t the GitHub credentials meant to act as an alternative to an email address for GitHub related tasks?</p>

**Comment by airtower-luna** [comment]
<aside class="quote no-group" data-username="0deadLock0" data-post="3" data-topic="152690">
<div class="title">
<div class="quote-controls"></div>
 0deadLock0:</div>
<blockquote>
<p>But, I wonder when someone is providing their GitHub credentials accurately why do GitHub specifically need the email address.</p>
</blockquote>
</aside>
<p>This is a general Git thing, not only for Github. Git records author information in the commit itself. For a bit more detail on that, see:</p><aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://git-scm.com/favicon.ico" class="site-icon" width="16" height="16">
      <a href="https://git-scm.com/book/en/v2/Git-Basics-Viewing-the-Commit-History" target="_blank" rel="noopener">git-scm.com</a>
  </header>
  <article class="onebox-body">
    <img src="" class="thumbnail" width="20" height="20">

<h3><a href="https://git-scm.com/book/en/v2/Git-Basics-Viewing-the-Commit-History" target="_blank" rel="noopener">Git - Viewing the Commit History</a></h3>



  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>It is possible (common, even) that people push commits they didn’t author, e.g. after merging code someone else contributed. Github cannot just ascribe commits to whoever pushes them, so they use the author information in the commits, which is meant for exactly that purpose.</p>
<p>If you (or your friend) want to keep your email address private, Github lets you use a special (not working for mail) address connected to your account. See the documentation <a href="https://docs.github.com/en/free-pro-team@latest/github/setting-up-and-managing-your-github-user-account/setting-your-commit-email-address#about-commit-email-addresses">About commit email addresses</a> for how to do that.</p>

**Comment by 0dminnimda** [comment]
<aside class="quote no-group" data-username="airtower-luna" data-post="4" data-topic="152690">
<div class="title">
<div class="quote-controls"></div>
 airtower-luna:</div>
<blockquote>
<p>If you (or your friend) want to keep your email address private, Github lets you use a special (not working for mail) address connected to your account. See the documentation <a href="https://docs.github.com/en/free-pro-team@latest/github/setting-up-and-managing-your-github-user-account/setting-your-commit-email-address#about-commit-email-addresses" rel="noopener nofollow ugc">About commit email addresses </a> for how to do that.</p>
</blockquote>
</aside>
<p>I found out that private emails are not shown in contributors (at least that happens with my account), so better to hold email public</p>

**Reply by Mbonnema2022** [reply]
i am having a similar issue i have a group project and my group collaberators pushes are not showing

**Comment by koblan1** [comment]
I have a question instead. I collaborated on a project with a friend. After pushing my side of the project, she no longer can see her commits and my GitHub account name alone is visible though the repo is in her GitHub account. What can we do to rectify the issue? We are both new to GitHub

**Reply by airtower-luna** [reply]
This is a very different question that should probably have its own thread.

The most likely thing I can think of is that you force-pushed without pulling her changes first and basically deleted her work. If that's what it is I hope she has a local copy so you can restore and merge properly.

If you make a new thread, make sure to describe how you set up your local repository and pushed, and if the repository is public include a link. :slightly_smiling_face:

**Reply by koblan1** [reply]
Thanks so much
